### PR TITLE
Leverage `dpctl.tensor.expand_dims()/swapaxes()` implementation

### PR DESCRIPTION
**Pull request by vlad-perevezentsev**
This PR modifies `dpnp.expand_dims` and `dpnp.swapaxes` implementations to use `dpctl.tensor.expand_dims()` and `dpctl.tensor.swapaxes()` instead of dpnp implementations. updates the tests for them and adds `swapaxes` method for `dpnp_array`.

- [X] Have you provided a meaningful PR description?
- [X] Have you added a test, reproducer or referred to issue with a reproducer?
- [X] Have you tested your changes locally for CPU and GPU devices?
- [X] Have you made sure that new changes do not introduce compiler warnings?
- [ ] Have you checked performance impact of proposed changes?
- [ ] If this PR is a work in progress, are you filing the PR as a draft?

### CODE REVIEW

File: tests/third_party/cupy/linalg_tests/test_eigenvalue.py [hunk 1]
```diff
@@ -8,15 +8,10 @@
...
     if UPLO == "U":
-        _a = _xp.triu(a) + _xp.triu(a, k=1).swapaxes(-2, -1).conj()
+        return xp.triu(a) + xp.triu(a, k=1).swapaxes(-2, -1).conj()
     else:
-        _a = _xp.tril(a) + _xp.tril(a, k=-1).swapaxes(-2, -1).conj()
-    return xp.array(_a)
+        return xp.tril(a) + xp.tril(a, k=-1).swapaxes(-2, -1).conj()
 
 
 # TODO: remove once all required functionality is supported
```
[the remark this answers is not in the document]
Comment (vlad-perevezentsev) [reply tests/third_party/cupy/linalg_tests/test_eigenvalue.py:17]: No cause `dpnp.dot` and `dpnp.matmul` do not support complex types yet

[the remark this answers is not in the document]
Comment (antonwolfy) [reply tests/third_party/cupy/linalg_tests/test_eigenvalue.py:17]: Could you please update the TODO comment to state explicitly what functions are still required?

[the remark this answers is not in the document]
Comment (vlad-perevezentsev) [reply tests/third_party/cupy/linalg_tests/test_eigenvalue.py]: Done

File: dpnp/dpnp_algo/dpnp_algo_manipulation.pxi [hunk 2]
```diff
@@ -104,35 +103,6 @@ cpdef utils.dpnp_descriptor dpnp_atleast_3d(utils.dpnp_descriptor arr):
...
-    cdef shape_type_c shape_list
-    axis_idx = 0
-    for i in range(result_ndim):
-        if i in axis_norm:
-            shape_list.push_back(1)
-        else:
-            shape_list.push_back(in_array.shape[axis_idx])
-            axis_idx = axis_idx + 1
-
-    return dpnp_reshape(in_array, shape_list)
```
Comment (antonwolfy) [inline dpnp/dpnp_algo/dpnp_algo_manipulation.pxi:133]: please remove `dpnp_reshape`

Comment (vlad-perevezentsev) [reply dpnp/dpnp_algo/dpnp_algo_manipulation.pxi:133]: done

**Review by antonwolfy — APPROVED** [review]
Great! Thank you @vlad-perevezentsev

**Pull request merged**